src/components/SectionTable/CourseInfoButton.js

```diff
@@ -1,26 +1,45 @@
-import React, { PureComponent } from 'react';
+import React from 'react';
 import PropTypes from 'prop-types';
 import { Button } from '@material-ui/core';
+import { useMediaQuery } from '@material-ui/core';
 
-class CourseInfoButton extends PureComponent {
-    render() {
-        const { text, icon, redirectLink } = this.props;
-        return (
-            <Button
-                startIcon={icon}
-                variant="contained"
-                size="small"
-                style={{ marginRight: '4px', backgroundColor: '#385EB1', color: '#fff' }}
-                onClick={(event) => {
-                    window.open(redirectLink);
-                }}
-            >
-                {text}
-            </Button>
-        );
-    }
+function CourseInfoButton({ text, icon, redirectLink }) {
+    const isMobileScreen = useMediaQuery('max-width: 750px)');
+
+    return (
+        <Button
+            startIcon={!isMobileScreen && icon}
+            variant="contained"
+            size="small"
+            style={{ marginRight: '4px', backgroundColor: '#385EB1', color: '#fff' }}
+            onClick={(event) => {
+                window.open(redirectLink);
+            }}
+        >
+            {text}
+        </Button>
+    );
 }
 
+//class CourseInfoButton extends PureComponent {
+//    render() {
+//        const { text, icon, redirectLink } = this.props;
+//        return (
+//            <Button
+//                startIcon={!isMobileScreen && icon}
+//                variant="contained"
+//                size="small"
+//                style={{ marginRight: '4px', backgroundColor: '#385EB1', color: '#fff' }}
+//                onClick={(event) => {
+//                    window.open(redirectLink);
+//                }}
+//            >
+//                {text}
+//            </Button>
+//        );
+//    }
+//}
+
 CourseInfoButton.propTypes = {
     text: PropTypes.string.isRequired,
     icon: PropTypes.object.isRequired,
```

Comment:
I don't think these prototypes work right now. See this link for how to do it with function components: https://reactjs.org/docs/typechecking-with-proptypes.html#function-components